pkg/controller/pvcontroller/sync.go

```diff
@@ -0,0 +1,195 @@
+	if k8serror.IsNotFound(err) {
+		runtime.HandleError(fmt.Errorf("PV %q has been deleted", key))
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// Deep-copy otherwise we are mutating cache store
+	err = pController.sync(pvObj.DeepCopy())
+	// TODO: If events needs to be generated undo the comments
+	// if err != nil {
+	// 	pController.Recorder.Event(pvObj, corev1.EventTypeWarning, "EventInformation", err.Error())
+	// }
+	return err
+}
+
+func (pController *PVMetricsController) sync(pvObj *corev1.PersistentVolume) error {
+	klog.V(4).Infof("Reconciling PV %s to send volume events", pvObj.Name)
+	if canSkipEventCollection(pvObj) {
+		return nil
+	}
+
+	klog.Infof("Got PV %s to send volume events", pvObj.Name)
+	eventSender, err := pController.getEventSender(pvObj)
+	if err != nil {
+		return err
+	}
+	if eventSender == nil {
+		return nil
+	}
+
+	// Send create information
+	if !isCreateVolumeEventSent(pvObj) {
+		// Get create event related data
+		data, err := eventSender.CollectCreateEventData()
+		if err != nil {
+			return errors.Wrapf(err, "failed to get create event data of volume %s", pvObj.Name)
+		}
+
+		// Send create event data
+		err = eventSender.Send(data)
+		if err != nil {
+			return errors.Wrapf(err, "failed to send create event data of volume %s to server", pvObj.Name)
+		}
+
+		_, err = eventSender.AnnotateCreateEvent(pvObj)
+		if err != nil {
+			return err
+		}
+		klog.Infof("Successfully sent create volume %s event to server", pvObj.Name)
+	}
+
+	// Send delete information
+	if pvObj.DeletionTimestamp != nil {
+		if !isDeleteVolumeEventSent(pvObj) {
+			// Get delete event related data
+			data, err := eventSender.CollectDeleteEventData()
+			if err != nil {
+				return errors.Wrapf(err, "failed to get delete event data of volume %s", pvObj.Name)
+			}
+
+			// Send delete event data
+			err = eventSender.Send(data)
+			if err != nil {
+				return errors.Wrapf(err, "failed to send delete event data of volume %s to server", pvObj.Name)
+			}
+
+			_, err = eventSender.AnnotateDeleteEvent(pvObj)
+			if err != nil {
+				return errors.Wrapf(err, "failed to annotate volume %s with delete event information", pvObj.Name)
+			}
+			klog.Infof("Successfully sent delete volume %s event to server", pvObj.Name)
+		}
+		err = eventSender.RemoveEventFinalizer()
+		if err != nil {
+			return errors.Wrapf(err, "failed to remove finalizers on volume %s", pvObj.Name)
+		}
+	}
+	return nil
+}
+
+func (pController *PVMetricsController) getEventSender(pvObj *corev1.PersistentVolume) (collectorinterface.EventsSender, error) {
+	var eventFinalizer string
+	for _, finalizer := range pvObj.Finalizers {
+		if strings.HasSuffix(finalizer, collectorinterface.OpenebsEventFinalizerSuffix) {
+			eventFinalizer = finalizer
+			break
+		}
+	}
+	if eventFinalizer == "" {
+		klog.Warningf("unable to find volume %s type from finalizers %v", pvObj.Name, pvObj.Finalizers)
+		return nil, nil
+	}
```

Comment:
Currently, we are sending events only if there is suffix `events.openebs.io/finalizer` in any one of the finalizers. I don't think this check is no longer required(since we are identifying volume type based on label[we can also do it by looking at provisioned name in PV annotation]). After getting the sender interface rest of the code flow should take care instead of returning here.